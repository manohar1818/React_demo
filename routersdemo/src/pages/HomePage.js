import React from 'react'
import { Link } from 'react-router-dom'
import Text from '../Atoms/Text/Text.js'


export default function HomePage() {
  return (
    <div className="container">
      {/*<h1>Home Page</h1>*/}
      <Text text="Home Page" type = "heading" />
      <p>
        <Link to="/manohar">manohar</Link> on GitHub.
      </p>
    </div>
  )
}