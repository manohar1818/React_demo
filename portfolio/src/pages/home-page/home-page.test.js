import React from 'react';
import { create } from "react-test-renderer";
import { render ,fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import HomePage from './home-page';
import { BrowserRouter } from 'react-router-dom';

describe("HomePage component",() =>{
    test("Matches the snapshot",() => {
        const input = create(<BrowserRouter>< HomePage/></BrowserRouter>);
        expect(input.toJSON()).toMatchSnapshot();

    })
    test("HomePage element is rendered",() => {
        const { getByTestId, getByText } = render(<BrowserRouter>< HomePage/></BrowserRouter>);
        expect(getByTestId("homePage")).toBeInTheDocument();
        
    });
   
   
   

});