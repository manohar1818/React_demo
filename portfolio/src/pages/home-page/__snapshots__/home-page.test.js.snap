// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component Matches the snapshot 1`] = `
<div
  data-testid="homePage"
>
  <div
    data-testid="headerTemplate"
  >
    <div
      className="makeStyles-styleHeader-1"
      data-testid="header"
    >
      <a
        className="makeStyles-link-2"
        href="/home"
        onClick={[Function]}
      >
        <h4
          className="MuiTypography-root makeStyles-styleHeaderElement-4 MuiTypography-h4"
          data-testid="headerElement"
        >
          Home
        </h4>
      </a>
      <a
        className="makeStyles-link-2"
        href="/gallery"
        onClick={[Function]}
      >
        <h4
          className="MuiTypography-root makeStyles-styleHeaderElement-4 MuiTypography-h4"
          data-testid="headerElement"
        >
          Gallery
        </h4>
      </a>
      <a
        className="makeStyles-link-2"
        href="/contact"
        onClick={[Function]}
      >
        <h4
          className="MuiTypography-root makeStyles-styleHeaderElement-4 MuiTypography-h4"
          data-testid="headerElement"
        >
          Contact
        </h4>
      </a>
      <a
        className="makeStyles-link-2"
        href="/about"
        onClick={[Function]}
      >
        <h4
          className="MuiTypography-root makeStyles-styleHeaderElement-4 MuiTypography-h4"
          data-testid="headerElement"
        >
          About
        </h4>
      </a>
      <a
        className="makeStyles-link-2"
        href="/gallery-admin"
        onClick={[Function]}
      >
        <h4
          className="MuiTypography-root makeStyles-styleHeaderElement-4 MuiTypography-h4"
          data-testid="headerElement"
        >
          GalleryAdmin
        </h4>
      </a>
      <div
        className="makeStyles-button-3"
      >
        <div
          className="makeStyles-button-5"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            data-testid="button"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="MuiTypography-root MuiTypography-button"
              >
                 
                Logout
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
